src/main/java/racingcar/GameSetter.java

```diff
@@ -0,0 +1,68 @@
+package racingcar;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class GameSetter {
+
+    private static final String NAME_SPLIT_DELIMITER = ",";
+    private static final Integer MIN_TRIAL = 1;
+    private static final Integer MAX_TRIAL = Integer.MAX_VALUE;
+    private static final int MIN_CARS_LENGTH = 2;
+
+    private GameSetter() {
+    }
+
+    public static List<Car> setCars(String input) {
+        String[] carNames = input.split(NAME_SPLIT_DELIMITER, -1);
+        validateCarsLength(carNames);
+        validateDuplication(carNames);
+        return Arrays.stream(carNames)
+            .map(Car::new)
+            .collect(Collectors.toList());
+    }
+
+    private static void validateCarsLength(String[] carNames) {
+        if (carNames.length < MIN_CARS_LENGTH) {
+            throw new IllegalArgumentException("자동차는 두 대 이상 입력해야 합니다.");
+        }
+    }
+
+    private static void validateDuplication(String[] carNames) {
+        HashSet<String> nameSet = new HashSet<>();
+        Collections.addAll(nameSet, carNames);
+        if (carNames.length != nameSet.size()) {
+            throw new IllegalArgumentException("중복되는 이름을 입력할 수 없습니다.");
+        }
+    }
+
+    public static Integer setTrial(String input) {
+        validateNumber(input);
+        try {
+            Integer trial = Integer.parseInt(input);
+            return validateBound(trial);
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException(
+                "숫자는 " + MIN_TRIAL + "이상 " + MAX_TRIAL + "이하여야 합니다.");
+        }
+    }
+
+    private static void validateNumber(String input) {
+        String numberPattern = "[0-9]+";
+        if (!Pattern.matches(numberPattern, input)) {
```

Comment:
정규식을 사용하셨네요 👍
해당메서드를 열어보시면 내부적으로 Pattern을 계속 생성하게 됩니다. 주석에도 아래와 같이 나와있어요.

> If a pattern is to be used multiple times, compiling it once and reusing it will be more efficient than invoking this method each time.

위와 같이 **패턴이 반복해서 사용된다면 미리 만들어놓고 재사용하는것이 더 효율적**이다라고 나와있습니다.
숫자를 나타내는 Pattern을 static 상수로 만들어놓고 재사용해보는건 어떨까요?